ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/ledger/inmemory/InMemoryLedger.scala

```diff
@@ -339,27 +340,19 @@ private[sandbox] final class InMemoryLedger(
   override def lookupMaximumLedgerTime(contractIds: Set[ContractId])(implicit
       loggingContext: LoggingContext
   ): Future[Option[Timestamp]] =
-    if (contractIds.isEmpty) {
-      Future.failed(
-        new IllegalArgumentException(
-          "Cannot lookup the maximum ledger time for an empty set of contract identifiers"
-        )
-      )
-    } else {
-      Future.fromTry(Try(this.synchronized {
-        contractIds
-          .foldLeft[Option[Instant]](Some(Instant.MIN))((acc, id) => {
-            val let = acs.activeContracts
-              .getOrElse(
-                id,
-                sys.error(s"Contract $id not found while looking for maximum ledger time"),
-              )
-              .let
-            acc.map(acc => if (let.isAfter(acc)) let else acc)
-          })
-          .map(Timestamp.assertFromInstant)
-      }))
-    }
+    Future.fromTry(Try(this.synchronized {
+      contractIds
+        .foldLeft[Option[Instant]](Some(Instant.MIN))((acc, id) => {
+          val let = acs.activeContracts
+            .getOrElse(
+              id,
+              throw MissingContracts(Set(id)),
```

Comment:
This exception will only ever contain a single contract id even if multiple are missing. 
I am ok with leaving it this way, as the in-memory ledger will disappear this year and it is not worth fixing it.

Reply:
Yes, this retains the same behavior as before. We only use the contractIds for logging. For the sake of the behavior, it's enough that the exception is thrown.